### PR TITLE
tests: Add end-to-end test setup

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal

Adds end-to-end tests to Laravel using Bugsnag-maze-runner.
Adds very basic unhandled test, to be expanded on in the future.

### Added
- Laravel test fixture
- Maze steps and environment to successfully run laravel test fixture

## Tests
Added test runs correctly, with correct responses.
Needs testing on another system to confirm correctness.

### CODE REVIEW

File: features/support/env.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'os'
+
+Before do
+  find_default_docker_compose
+end
+
+def install_current_branch(fixture)
```
Comment (snmaynard) [inline features/support/env.rb]: This is brittle as it wont test local changes. Why not use the path options to composer -> https://getcomposer.org/doc/05-repositories.md#path

File: features/steps/laravel_steps.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+require 'net/http'
+
+When("I configure the bugsnag endpoint") do
```
Comment (snmaynard) [inline features/steps/laravel_steps.rb]: Most, if not all of these, should be in maze itself imo.

File: features/support/env.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  branches = `git branch`.split("\n")
+  current_branch = branches.map{ |line| /\*\s(.*)/.match(line)}.compact.first
+  branch_name = current_branch.captures.first
+  branch_name.sub! '/', '\/'
+  Dir.chdir "features/fixtures/#{fixture}" do
+    `cat composer.json.base | sed 's/TEST_BRANCH_NAME/#{branch_name}/g' > composer.json`
+  end
+end
+
+def current_ip
```
Comment (snmaynard) [inline features/support/env.rb]: This should be in maze as its also used by ruby at least.

File: features/support/env.rb (hunk 1 above)
Comment (Cawllec) [reply features/support/env.rb]: Composer won't install a parent directory as a package, which it would be in this case.

Comment (snmaynard) [reply features/support/env.rb]: why dont you cp into another directory so its not parent?

**Pull request merged**